### PR TITLE
feat(home): add cookies dialog

### LINKED ISSUE #188: add popup for cookies

### PR DESCRIPTION
**Pull request by Iqro-dev**
closes #188

### CODE REVIEW

File: app/hooks/use-auth-cookies.tsx [hunk 1]
```diff
@@ -1,25 +1,34 @@
...
     REFRESH_TOKEN,
     USER_ID,
+    USER_ACCEPT_COOKIES,
   ])
 
   return {
     accessToken: cookie[ACCESS_TOKEN] as string,
     refreshToken: cookie[REFRESH_TOKEN] as string,
     userId: cookie[USER_ID] as string,
+    userAcceptCookies: cookie[USER_ACCEPT_COOKIES] as boolean,
```
Comment (Mnigos) [inline app/hooks/use-auth-cookies.tsx:22]: This is not auth cookie. just use `useCookies` hook in dialog component

**Review by Mnigos — CHANGES_REQUESTED** [review]

**Comment by Mnigos** [comment]
This popup should appear on all pages not only on home. User shouldn't be able to access his profile without accepting cookies policy.

File: app/cookies-dialog.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
+'use client'
```
Comment (Mnigos) [inline app/cookies-dialog.tsx:1]:
```suggestion
'use client'

```

File: app/layout.tsx [hunk 3]
```diff
@@ -20,6 +21,7 @@ export const metadata = {
 export default async function RootLayout({ children }: LayoutProps) {
   const accessToken = cookies().get(ACCESS_TOKEN)?.value
   const userId = cookies().get(USER_ID)?.value
+  const isAccepted = cookies().has(USER_ACCEPT_COOKIES as string)
```
Comment (Mnigos) [inline app/layout.tsx:24]: why `as string`?

**Review by Mnigos — APPROVED** [review]

File: app/layout.tsx (hunk 3 above)
Comment (Iqro-dev) [reply app/layout.tsx:24]: because it throws an error "unsafe assignment of any"

**Pull request merged**